packages/editor-ui/src/components/NodeCredentials.vue

```diff
@@ -134,15 +171,100 @@ export default mixins(genericHelpers, nodeHelpers, restApi, showMessage).extend(
 		return {
 			NEW_CREDENTIALS_TEXT: `- ${this.$locale.baseText('nodeCredentials.createNew')} -`,
 			subscribedToCredentialType: '',
+			listeningForAuthChange: false,
 		};
 	},
 	mounted() {
-		this.listenForCredentialUpdates();
+		// Listen for credentials store changes so credential selection can be updated if creds are changed from the modal
+		this.credentialsStore.$onAction(({ name, after, store, args }) => {
+			const listeningForActions = ['createNewCredential', 'updateCredential', 'deleteCredential'];
+			const credentialType = this.subscribedToCredentialType;
+			if (!credentialType) {
+				return;
+			}
+
+			after(async (result) => {
+				if (listeningForActions.includes(name)) {
+					const current = this.selected[credentialType];
+					let credentialsOfType: ICredentialsResponse[] = [];
+					if (this.showAll) {
+						if (this.node) {
+							credentialsOfType = [
+								...(this.credentialsStore.allUsableCredentialsForNode(this.node) || []),
+							];
+						}
+					} else {
+						credentialsOfType = [
+							...(this.credentialsStore.allUsableCredentialsByType[credentialType] || []),
+						];
+					}
+					switch (name) {
+						// new credential was added
+						case 'createNewCredential':
+							if (result) {
+								this.onCredentialSelected(credentialType, (result as ICredentialsResponse).id);
+							}
+							break;
+						case 'updateCredential':
+							const updatedCredential = result as ICredentialsResponse;
+							// credential name was changed, update it
+							if (updatedCredential.name !== current.name) {
+								this.onCredentialSelected(credentialType, current.id);
+							}
+							break;
+						case 'deleteCredential':
+							// all credentials were deleted
+							if (credentialsOfType.length === 0) {
+								this.clearSelectedCredential(credentialType);
+							} else {
+								const id = args[0].id;
+								// credential was deleted, select last one added to replace with
+								if (current.id === id) {
+									this.onCredentialSelected(
+										credentialType,
+										credentialsOfType[credentialsOfType.length - 1].id,
+									);
+								}
+							}
+							break;
+					}
+				}
+			});
+		});
+	},
+	watch: {
+		'ndvStore.activeNode.parameters': {
```

Comment:
cannot we just use node prop here?

Reply:
Not sure. Auth field name is not the same for all nodes, so here I am watching all params changes and figuring out if auth type has changed, if that was the question.

Reply:
is not `ndvStore.activeNode` same as the prop `node` here?